packages/next/build/webpack-config.ts

```diff
@@ -248,20 +250,22 @@ export default async function getBaseWebpackConfig(
             )
           },
           name(module: { identifier: Function; rawRequest: string }): string {
-            const rawRequest =
-              module.rawRequest &&
-              module.rawRequest.replace(/^@(\w+)[/\\]/, '$1-')
-            if (rawRequest) return rawRequest
-
             const identifier = module.identifier()
+            // Remove everything up through '/node_modules/'
             const trimmedIdentifier = /(?:^|[/\\])node_modules[/\\](.*)/.exec(
               identifier
             )
             const processedIdentifier =
               trimmedIdentifier &&
-              trimmedIdentifier[1].replace(/^@(\w+)[/\\]/, '$1-')
+              // Remove the file extension(s)
+              /[\w-\/]+/.exec(trimmedIdentifier[1])
+
+            let finalName = processedIdentifier && processedIdentifier[0]
+
+            finalName = finalName && finalName.split('/').join('~')
+            const backupName = identifier.split('/').join('~')
```

Comment:
Not sure if the right move here is to fail gracefully (as here) or to throw. Going with the former for now, unless someone objects

Reply:
Can we make this just the hashed module ID? I'm not sure there's value in trying to provide a name.

Reply:
Turns out it wasn't needed after all

Reply:
Why the module ID and not just the file name, as here? If it's a security issue (ie we're worried about exposing any information about the file structure of the machine that built the app) then we'd have to use something other than the fast SHA1 we're using for hashes. I tend not to think that would actually be a security issue, though, and this backup shouldn't actually get used for any properly formed module filepath.

Reply:
Just so we can shed all of this logic stripping `node_modules`, etc. Webpack already generates an ID for us so I figured it'd be better to rely on it.

Reply:
I'm not exactly sure why, but in the `name()` callback, `module.id` has a value of null. The `module.identifier()` method returns the file path. Was there a different way you were thinking of to access an ID?

Reply:
Hmm, `module.id` is dependent on the chunk graph so the ID I'd like to use isn't available yet. 😞 

Looks like we could call `module.libIdent({ context: dir })` though instead of `module.identifier()` and get an already-relative path.

Maybe we could replace all of this with something like:
```js
	name(module: { libIdent: Function }): string {
		return crypto
			.createHash("sha1")
			.update(module.libIdent({ context: dir }))
			.digest("hex")
			.substring(0, 8);
	}
```